core/src/main/java/org/mule/runtime/core/el/DataWeaveExpressionLanguage.java

```diff
@@ -123,14 +139,19 @@ private void addEventBindings(Event event, BindingContext.Builder contextBuilder
         contextBuilder.addBinding(name, value);
       });
       contextBuilder.addBinding(VARIABLES,
-                                new TypedValue(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
+                                new TypedValue<>(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
       Message message = event.getMessage();
       Attributes attributes = message.getAttributes();
-      contextBuilder.addBinding(ATTRIBUTES, new TypedValue(attributes, fromType(attributes.getClass())));
+      contextBuilder.addBinding(ATTRIBUTES, new TypedValue<>(attributes, fromType(attributes.getClass())));
       contextBuilder.addBinding(PAYLOAD, message.getPayload());
-      contextBuilder.addBinding(DATA_TYPE, new TypedValue(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(DATA_TYPE, new TypedValue<>(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(MULE,
+                                new TypedValue<>(new MuleInstanceContext(muleContext), fromType(MuleInstanceContext.class)));
+      contextBuilder.addBinding(SERVER, new TypedValue<>(new ServerContext(), fromType(ServerContext.class)));
+      contextBuilder.addBinding(APP, new TypedValue<>(new AppContext(muleContext), fromType(AppContext.class)));
```

Comment:
The AppContext is what we used to expose in mvel. And currently there is no way to filter. Pablo LG said that for now we do this and later when spring is revisited they are going to filter it